pulsar/producer_partition.go

```diff
@@ -742,6 +747,10 @@ func (p *partitionProducer) internalSendAsync(ctx context.Context, msg *Producer
 		callback(nil, msg, errProducerClosed)
 		return
 	}
+	if p.err != nil {
```

Comment:
This now changes the current behavior. Before if there was room the message would be added to the pending queue and sent when a new connection is established. Now the message will not be added. I'm not sure changing the default behavior is the correct approach here.

How do you envision your application handling this error?